### ISSUE TITLE
Could not obtain current Hibernate Session; nested exception is org.hibernate.HibernateException: No Session found for current thread

### ISSUE THREAD

**Issue by VenelinPetkov** [grails-plugins/grails-spring-security-core#378]
Grails 3.0.7 and Spring Security fail to authenticate

```
ERROR grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter - An internal error occurred while trying to authenticate the user.
org.springframework.security.authentication.InternalAuthenticationServiceException: Could not obtain current Hibernate Session; nested exception is org.hibernate.HibernateException: No Session found for current thread
	at org.springframework.security.authentication.dao.DaoAuthenticationProvider.retrieveUser(DaoAuthenticationProvider.java:110) ~[spring-security-core-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.authentication.dao.AbstractUserDetailsAuthenticationProvider.authenticate(AbstractUserDetailsAuthenticationProvider.java:132) ~[spring-security-core-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.authentication.ProviderManager.authenticate(ProviderManager.java:156) ~[spring-security-core-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter.attemptAuthentication(UsernamePasswordAuthenticationFilter.java:92) ~[spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.attemptAuthentication(RequestHolderAuthenticationFilter.groovy:73) [spring-security-core-3.0.0.M1.jar:na]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:211) ~[spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.groovy:53) [spring-security-core-3.0.0.M1.jar:na]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.groovy:62) [spring-security-core-3.0.0.M1.jar:na]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:73) [grails-web-mvc-3.0.7.jar:3.0.7]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67) [grails-web-mvc-3.0.7.jar:3.0.7]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:85) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:68) [spring-boot-actuator-1.2.5.RELEASE.jar:1.2.5.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:142) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:518) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1091) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:668) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1521) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1478) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_60]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_60]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-embed-core-8.0.23.jar:8.0.23]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_60]
```

**Comment by alvarosanchez** [comment]
Could you please indicate how to reproduce the issue?

**Comment by VenelinPetkov** [comment]
Hello,

Init a new graild 3.0.7 project, then add the security plugin, simple controller with check if the user is logged in or not (    if (springSecurityService.isLoggedIn()) {.....) and it is failing.

I will give it a try with 3.0.8, if it is grails related and let you know.

**Comment by VenelinPetkov** [comment]
I have checked it with 3.0.8 and the behaviour is the same.

Same error message:
ERROR grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter - An internal error occurred while trying to authenticate the user.
org.springframework.security.authentication.InternalAuthenticationServiceException: Could not obtain current Hibernate Session; nested exception is org.hibernate.HibernateException: No Session found for current thread
	at org.springframework.security.authentication.dao.DaoAuthenticationProvider.retrieveUser(DaoAuthenticationProvider.java:110) ~[spring-security-core-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.authentication.dao.AbstractUserDetailsAuthenticationProvider.authenticate(AbstractUserDetailsAuthenticationProvider.java:132) ~[spring-security-core-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.authentication.ProviderManager.authenticate(ProviderManager.java:156) ~[spring-security-core-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter.attemptAuthentication(UsernamePasswordAuthenticationFilter.java:92) ~[spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.attemptAuthentication(RequestHolderAuthenticationFilter.groovy:73) [spring-security-core-3.0.0.M1.jar:na]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:211) ~[spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.groovy:53) [spring-security-core-3.0.0.M1.jar:na]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.groovy:62) [spring-security-core-3.0.0.M1.jar:na]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160) [spring-security-web-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:73) [grails-web-mvc-3.0.8.jar:3.0.8]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67) [grails-web-mvc-3.0.8.jar:3.0.8]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:85) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:69) [spring-boot-actuator-1.2.6.RELEASE.jar:1.2.6.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:142) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:518) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1091) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:673) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1526) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1482) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_60]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_60]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-embed-core-8.0.26.jar:8.0.26]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_60]
Caused by: org.springframework.dao.DataAccessResourceFailureException: Could not obtain current Hibernate Session; nested exception is org.hibernate.HibernateException: No Session found for current thread
	at org.grails.orm.hibernate.GrailsHibernateTemplate.getSession(GrailsHibernateTemplate.java:229) ~[grails-datastore-gorm-hibernate4-4.0.6.RELEASE.jar:na]
	at org.grails.orm.hibernate.GrailsHibernateTemplate.doExecute(GrailsHibernateTemplate.java:185) ~[grails-datastore-gorm-hibernate4-4.0.6.RELEASE.jar:na]
	at org.grails.orm.hibernate.GrailsHibernateTemplate.execute(GrailsHibernateTemplate.java:142) ~[grails-datastore-gorm-hibernate4-4.0.6.RELEASE.jar:na]
	at org.grails.orm.hibernate.GrailsHibernateTemplate.execute(GrailsHibernateTemplate.java:112) ~[grails-datastore-gorm-hibernate4-4.0.6.RELEASE.jar:na]
	at org.grails.orm.hibernate.AbstractHibernateGormStaticApi.findWhere(AbstractHibernateGormStaticApi.groovy:335) ~[grails-datastore-gorm-hibernate-core-4.0.6.RELEASE.jar:na]
	at org.grails.datastore.gorm.GormStaticApi.findWhere(GormStaticApi.groovy:629) ~[grails-datastore-gorm-4.0.6.RELEASE.jar:na]
	at org.grails.datastore.gorm.GormStaticApi$findWhere$2.call(Unknown Source) ~[na:na]
	at org.grails.datastore.gorm.GormEntity$Trait$Helper.findWhere(GormEntity.groovy:656) ~[grails-datastore-gorm-4.0.6.RELEASE.jar:na]
	at org.grails.datastore.gorm.GormEntity$Trait$Helper$findWhere$13.call(Unknown Source) ~[na:na]
	at service.admin.security.User.findWhere(User.groovy) ~[main/:na]
	at service.admin.security.User$findWhere$5.call(Unknown Source) ~[na:na]
	at grails.plugin.springsecurity.userdetails.GormUserDetailsService.$tt__loadUserByUsername(GormUserDetailsService.groovy:60) ~[spring-security-core-3.0.0.M1.jar:na]
	at grails.plugin.springsecurity.userdetails.GormUserDetailsService$_loadUserByUsername_closure1.doCall(GormUserDetailsService.groovy) ~[spring-security-core-3.0.0.M1.jar:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_60]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_60]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_60]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_60]
	at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodInvoke(ReflectiveInterceptor.java:1426) ~[springloaded-1.2.4.RELEASE.jar:1.2.4.RELEASE]
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93) ~[groovy-2.4.4.jar:2.4.4]
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325) ~[groovy-2.4.4.jar:2.4.4]
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294) ~[groovy-2.4.4.jar:2.4.4]
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1019) ~[groovy-2.4.4.jar:2.4.4]
	at groovy.lang.Closure.call(Closure.java:426) ~[groovy-2.4.4.jar:2.4.4]
	at groovy.lang.Closure.call(Closure.java:442) ~[groovy-2.4.4.jar:2.4.4]
	at grails.transaction.GrailsTransactionTemplate$2.doInTransaction(GrailsTransactionTemplate.groovy:93) ~[grails-core-3.0.8.jar:3.0.8]
	at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:133) ~[spring-tx-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at grails.transaction.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:90) ~[grails-core-3.0.8.jar:3.0.8]
	at grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy) ~[spring-security-core-3.0.0.M1.jar:na]
	at grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:71) ~[spring-security-core-3.0.0.M1.jar:na]
	at org.springframework.security.authentication.dao.DaoAuthenticationProvider.retrieveUser(DaoAuthenticationProvider.java:102) ~[spring-security-core-3.2.8.RELEASE.jar:3.2.8.RELEASE]
	... 46 common frames omitted
Caused by: org.hibernate.HibernateException: No Session found for current thread
	at org.grails.orm.hibernate.GrailsSessionContext.currentSession(GrailsSessionContext.java:117) ~[grails-datastore-gorm-hibernate4-4.0.6.RELEASE.jar:na]
	at org.hibernate.internal.SessionFactoryImpl.getCurrentSession(SessionFactoryImpl.java:1014) ~[hibernate-core-4.3.11.Final.jar:4.3.11.Final]
	at org.grails.orm.hibernate.SessionFactoryProxy.getCurrentSession(SessionFactoryProxy.java:148) ~[grails-datastore-gorm-hibernate4-4.0.6.RELEASE.jar:na]
	at org.grails.orm.hibernate.GrailsHibernateTemplate.getSession(GrailsHibernateTemplate.java:227) ~[grails-datastore-gorm-hibernate4-4.0.6.RELEASE.jar:na]
	... 75 common frames omitted


I would say I did not make much changes from the base project, e.g:

build.gradle..
....
        compile "org.grails.plugins:spring-security-core:3.0.0.M1"
...
Bootstrap.groovy...
...

                UserRole.findAll().each { it.delete() }
                Role.findAll().each { it.delete() }
                User.findAll().each { it.delete() }

                def adminRole = new Role('ROLE_ADMIN').save()
                def adminR = new Role('ADMIN').save()
                def userRole = new Role('ROLE_USER').save()
                def userR = new Role('USER').save()

                def testUser = new User('test', 'test').save()

                UserRole.create testUser, adminRole, true
                UserRole.create testUser, adminR, true
                UserRole.create testUser, userR, true
                UserRole.create testUser, userRole, true

                assert User.count() == 1
                assert Role.count() == 4
                assert UserRole.count() == 4
...

simple controller:

class PageController {

    def springSecurityService

    def index() {
        log.info "Action['$actionName']: Params: $params"
        if (springSecurityService.isLoggedIn()) {
            redirect(controller: 'dashboard', actin: 'index')
        } else {
            redirect(controller: 'login', actin: 'index')
        }
    }
}

and when login to the form, the error message is thrown...

**Comment by e210** [comment]
same error in 3.0.9
it works only in run-app

**Comment by yancovski** [comment]
I had this issue with custom UserDetailsService, my solution was  to  mark the service as @Transactional

**Comment by smakela** [comment]
This is weird, but I get this error when adding mongo -plugin in to the project.
I attached grails 3.0.11 app with basic tutorial steps done.
Only difference is that I included mongo 4.0.7 plugin to it.

Just try to login and you will get the Hibernate error.

[security-mongo.tar.gz](https://github.com/grails-plugins/grails-spring-security-core/files/109908/security-mongo.tar.gz)

**Comment by tedder** [comment]
My error is in 2.5, not 3.x, but if the same pattern holds, you can wrap your hibernate calls like this:

     SomeDomain.withTransaction { ->
        // code here
    }

See Burt's answer here:
https://stackoverflow.com/questions/34095120/accessing-session-from-a-filter

**Comment by burtbeckwith** [comment]
@smakela We were just discussing problems with Hibernate+MongoDB last night on Slack. It's a bug in GORM that's been around for a while, completely unrelated to the plugin. I uploaded a test app that shows the problem here: https://github.com/burtbeckwith/mongo_broken_sessions - it works fine with just Hibernate, but adding the MongoDB plugin (with or without actually using MongoDB) causes `getCurrentSession` to fail

**Comment by smakela** [comment]
@burtbeckwith Is there any workaround? I'm just trying to migrate all our plugins and apps to 3.x. And we really need mongo plugin to work.

**Comment by smakela** [comment]
Ok. I just tested your app and it was 3.1 application, so it's not fixed in new GORM implementation. Is there going to be fix soon?

**Comment by burtbeckwith** [comment]
Graeme is taking a look, and I created https://github.com/grails/grails-data-mapping/issues/634 to track progress. Please discuss on Slack in the #gorm channel if you come across anything that would help resolve the issue

**Comment by mlim1972** [comment]
I was upgrading my application from 2.4.3 to 2.5.3 and found the same issue.  Then I rolled back to the previous hibernate I was using in 2.4.3 and things started to work back again.  For now, I'll keep the older hibernate version as a workaround.  As reference, I tried all the versions since 2.4.3 to 2.5.1/2.53 and here are the results:

// version from 2.4.3 - WORKS:
runtime ':hibernate4:4.3.5.5' // or ':hibernate:3.6.10.17'

// version from 2.4.4 - DOES NOT WORK:
runtime ':hibernate4:4.3.6.1' // or ':hibernate:3.6.10.18'

// version from 2.4.5 / 2.5.0 - DOES NOT WORK:
runtime ':hibernate4:4.3.8.1' // or ':hibernate:3.6.10.19'

// version from 2.5.1 - DOES NOT WORK:
runtime ':hibernate4:4.3.10' // or ':hibernate:3.6.10.19'

**Comment by aspino** [comment]
This issue also seems to be affecting our Wildfly app deployments that are using just hibernate/GORM 5.0 (not including anything related to mongoDB).

You can check a sample Grails 3.1 application with Spring Security Plugin 3.0.3 exhibiting the problem  here: https://github.com/rmondejar/grails3-library-example. Just build a war and try to deploy it on Wildfly 10. This is currently using the latest version of hibernate 4 but going back to previous versions does not avoid the `No Session found for current thread` error.

**Comment by mlim1972** [comment]
@aspino , that's correct, my application is also not using mongodb plugin and I'm getting the same error.  I'm using Grails 2.5.3 + Spring Security Core 2.0 and same issue.  Changing the hibernate version to 4.3.5.5 seems to have workaround the issue though.

**Comment by aspino** [comment]
@mlim1972 After reading your first comment, I also tried with Hibernate 4.3.5.Final, but unfortunately the problem persists. However, keep in mind I'm using Grails 3.1 + Spring Security Plugin 3.0.3, so it's possible that I might be facing an entirely different problem that just happens to raise the same exception.

Thanks for the suggestion though!

**Comment by leleFc** [comment]
Same problem with grails 3.1.4 and Spring security 3.0.4 on IBM WAS.
Everything is ok if I use Tomcat 8 instead WAS.
I'm using JDK 1.7

**Comment by graemerocher** [comment]
@aspino I have done some investigating and so far what I have found is that the `No Session` problem exists in WildFly 8.2.1 but not in WildFly 10.0.0 

I used the `grails3-library-example` app and these steps to deploy: http://grails.io/post/142674392718/deploying-grails-3-to-wildfly-10

This indicates to me that the problem is with the container implementation and not Grails itself, so if it is possible to upgrade your container that is the recommended path forward. However, I will continue to dig to try and find out why exactly it is manifesting in 8.2.1

**Comment by aspino** [comment]
@graemerocher We're already using Wildfly 10, but keep seeing the error nonetheless. I tried adding the `org.hibernate.validator` exclusion as suggested in your blog post, but the `No Session found for current thread` persists. In fact we've never experienced any problem with Hibernate Validator in Wildfly 10, which is weird...

One question: did you actually try to authenticate? That's when the error appears. Otherwise the app deploys and seems to work fine.

Thanks for looking into this!

**Comment by graemerocher** [comment]
@aspino yes authentication work for me in WildFly 10 but not WildFly 8

I have debugged further and it seems that the Spring `transactionManager` is not a `HibernateTransactionManager` but a plain `DataSourceTransactionManager`, which is the root cause of the issue. I think it has something to do with JTA container managed transactions replacing the hibernate one which is causing problems.

**Comment by graemerocher** [comment]
Ok the issue is with Spring Boot replacing the transaction manager with a JTA container managed one:

https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/transaction/jta/JtaAutoConfiguration.java

Probably we should handle the case where there is a JTA transaction manager better, but in the meantime the issue can be resolved by modifying your `Application` class as follows:

```
import org.springframework.boot.autoconfigure.transaction.jta.*
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import grails.boot.GrailsApp
import grails.boot.config.GrailsAutoConfiguration

@EnableAutoConfiguration(exclude=[JtaAutoConfiguration])
class Application extends GrailsAutoConfiguration {
    static void main(String[] args) {
        GrailsApp.run(Application, args)
    }
}
```

This effectively prevents Spring Boot from configuration a `JtaTransactionManager` that replaces the `HibernateTransactionManager` that Grails configures.

**Comment by burtbeckwith** [comment]
Although this appears to be related to the plugin, in all cases above it appears that that is not the case. Users are seeing session-related exceptions when making GORM calls in the plugin, but it's a coincidence - authentication is often the first database-access call made but anything related to Hibernate's current session would trigger this.

If anyone continues to see problems and it really seems to be related to this plugin and not a generic Hibernate/GORM issue please create a new issue.

**Comment by graemerocher** [comment]
@aspino I have written a blog which details correctly configuring Grails/GORM on WildFly:

http://grails.io/post/142674392718/deploying-grails-3-to-wildfly-10

We could probably add support for JTA/container managed transactions in the future, but right now you should use Grails/Spring managed transactions as described in my above blog post to avoid issues like this.

**Comment by aspino** [comment]
Thanks @graemerocher. Disabling JTA autoconfiguration did the trick.

**Comment by joergrech** [comment]
I still get this error with Grails 3.2.0.M1 and springsecurity 3.1.1 without mongo. I'm trying to migrate a Grails 2.4.4 project to Grails 3. Excluding the JtaAutoConfiguration (in Application.groovy) did not help either.

**Comment by bilgehanm** [comment]
I get this error while upgrading a Grails 2.4.5 project to Grails 3.1.11.

**Comment by edwardotis** [comment]
@joergrech Did you find a workaround?

I'm experiencing the same issue trying to upgrade to Grails 2.5.5 to Grails 3.1.13,  springsecurity 3.1.1 without mongo. org.grails.plugins:hibernate4:5.0.12

Disabling JTA autoconfiguration had no effect.

**Comment by joergrech** [comment]
No, we upgraded to 2.5.5 and will revisit Grails 3 next year.

**Comment by edwardotis** [comment]
ok, thanks. I have tried numerous combinations of grails 3.1.x, hibernate4.x, postgres, and spring dependencies and configurations. I even started over with a simple dummy project to test the minimal configuration, and I can't get rid of the error: org.hibernate.HibernateException: No Session found for current thread

If I make a controller that is set for 'permitAll' to bypass spring-security-core, it has no problem with hibernate and gorm access, so I am pretty certain it's a problem with grails-spring-security-core

I'm going to revert back to 2.5.5 as well. 

I wish there was an example app on github with a functioning configuration for grails-spring-security-core and grails 3.1.x

**Comment by edwardotis** [comment]
Update:
The org.jadira.usertype:usertype.extended  dependency was the root cause of the migration problems and hibernate errors in my case. I've resolved that issue, and now spring-security-core is able to get a hibernate session.

TL;DR:
Exclude 'hibernate-entitymanager'
```
compile('org.jadira.usertype:usertype.extended:4.0.0.GA') {
exclude group: 'org.hibernate', module: 'hibernate-entitymanager'
}
```

This bug has the details:
#https://github.com/grails/grails-core/issues/9304
